docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,704 @@
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            // The entities to add the spawn costs for
+            entities.getOrThrow(EntityTypeTags.SKELETONS),
+            new MobSpawnSettings.MobSpawnCost(
+                1.0, // The energy budget
+                0.1  // The amount of charge each entity takes up from the budget
+            )
+        )
+    );
+});
+```
+
+  </TabItem>
+</Tabs>
+
+
+### Remove Spawn Costs
+
+Allows for removing a spawn cost from a biome. Spawn costs are a newer way of making mobs spawn spread out in a biome to reduce clustering. The modifier takes in the biome id or tag of the biomes the spawn costs are removed from, and the `EntityType` id or tag of the entities to remove the spawn cost for.
+
+<Tabs>
+  <TabItem value="json" label="JSON" default>
+
+```json5
+{
+  "type": "neoforge:remove_spawn_costs",
+  // Can either be a biome id, such as "minecraft:plains",
+  // or a list of biome ids, such as ["minecraft:plains", "minecraft:badlands", ...],
+  // or a biome tag, such as "#c:is_overworld".
+  "biomes": "#namespace:biome_tag",
+  // Can either be an entity type id, such as "minecraft:ghast",
+  // or a list of entity type ids, such as ["minecraft:ghast", "minecraft:skeleton", ...],
+  // or an entity type tag, such as "#minecraft:skeletons".
+  "entity_types": "#minecraft:skeletons"
+}
+```
+
+  </TabItem>
+  <TabItem value="datagen" label="Datagen">
+
+```java
+// Define the ResourceKey for our BiomeModifier.
+public static final ResourceKey<BiomeModifier> REMOVE_SPAWN_COSTS_EXAMPLE = ResourceKey.create(
+    NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+    ResourceLocation.fromNamespaceAndPath(MOD_ID, "remove_spawn_costs_example") // The registry name
+);
+
+// BUILDER is a RegistrySetBuilder passed to DatapackBuiltinEntriesProvider
+// in a listener for GatherDataEvent.
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries.
+    // Static registries only need to be looked up if you need to grab the tag data.
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+    HolderGetter<EntityType<?>> entities = bootstrap.lookup(Registries.ENTITY_TYPE);
+
+    // Register the biome modifiers.
+    bootstrap.register(REMOVE_SPAWN_COSTS_EXAMPLE,
+        new RemoveSpawnCostsBiomeModifier(
+            // The biome(s) to remove the spawn costs from
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            // The entities to remove spawn costs for
+            entities.getOrThrow(EntityTypeTags.SKELETONS)
+        )
+    );
+});
+```
+
+  </TabItem>
+</Tabs>
+
+
+### Add Legacy Carvers
+
+This biome modifier type allows adding carver caves and ravines to biomes. These are what was used for cave generation before the Caves and Cliffs update. It CANNOT add noise caves to biomes, because noise caves are baked into the dimension's noise settings system and not actually tied to biomes.
```

Comment:
Do we need to be more clear that it is for dimensions that use the noise-based chunk generator, or is it generally assumed most dimensions will use this?

Reply:
I am not sure what you mean. This is about legacy carvers so the focus is on them. Legacy carvers can be added to any dimension by adding them to biome. If a person wants to add noise caves, they would need to look those up elsewhere as we state here that this isn't the right tool for those kinds of caves

Reply:
I was talking about 'noise caves are baked in the dimension's noise settings', which is not technically true as it's from the noise-based chunk generator most dimensions use. I was asking more if we should care about the minor simplification there.